### PR TITLE
Add abstract interface manager

### PR DESCRIPTION
**Pull request by koesie10**
This will add a new abstract class that implements the creation of the panel and webview to reduce duplication across the different interface managers.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by aeisenberg — COMMENTED** [review]
Quick review. Haven't tried this out yet. Looks good so far.

File: extensions/ql-vscode/src/abstract-interface-manager.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return !!this.panel;
+  }
+
+  protected getPanel(): WebviewPanel {
+    if (this.panel == undefined) {
+      const { ctx } = this;
+
+      const config = this.getPanelConfig();
+
+      const panel = (this.panel = Window.createWebviewPanel(
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/src/abstract-interface-manager.ts:46]: Thanks, that is indeed better

File: extensions/ql-vscode/src/abstract-interface-manager.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        {
+          enableScripts: true,
+          enableFindWidget: true,
+          retainContextWhenHidden: true,
+          ...config.additionalOptions,
+          localResourceRoots: [
+            ...(config.additionalOptions?.localResourceRoots ?? []),
+            Uri.file(tmpDir.name),
+            Uri.file(path.join(ctx.extensionPath, 'out')),
+            Uri.file(path.join(ctx.extensionPath, 'node_modules/@vscode/codicons/dist')),
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/src/abstract-interface-manager.ts:59]: You might have gotten this from the previous PR already, but it's indeed correct that we are now loading it for all the views.

**Review by aeisenberg — APPROVED** [review]
Tested this locally. Seems to be working. LGTM.

**Pull request merged**